### DISCUSSION TITLE
Compiling pyOFM on HPC system

### DISCUSSION THREAD

**Discussion by uttamcadambi07** [mdolab/dafoam#374] [Installation]
Dear,

Please find the link to the issue in building pyOFM on HPC that was posted https://github.com/mdolab/pyofm/issues/10

Please let me know if there is a fix to this.

**Comment by friedenhe** [comment]
Check how to compile DAFoam and its dependencies on HPC from here: https://dafoam.github.io/mydoc_installation_source_hpc.html. You are not supposed to load both Gcc and Intel compilers...

**Reply by uttamcadambi07** [reply]
Thank you Dr. He.

I have consulted the very same link and I have been building it on my HPC cluster for over a week now. Of course, I made all the adaptations for my particular system. I managed to have everything up until pyOFM installed and working correctly.

For this error, I initially got this

```
File "setup.py", line 38, in <module>
    os.getenv("FOAM_SRC") + "/meshTools/lnInclude",
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
make: *** [default] Error 1
```

Then I did: 

```
export FOAM_SRC=$FOAM_SRC:/scratch/ucp1n22/DAFoam/OpenFOAM/OpenFOAM-v1812/src
```
To set the path to the openfoam source folder.

Once i tried to do "make" after doing these steps in pyOFM folder, i was faced with the error that I posted on the pyOFM forum. 

I just want to know if the error in this line

```
File "/local/software/python/3.7.3/lib/python3.7/posixpath.py", line 80, in join
    a = os.fspath(a)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

could be because of something we missed in the build process or of the source code. Because we followed every single step of the way and managed to come this far. Any input from your end will be extremely helpful to us. 

Please also let us know if you need any additional information.

**Reply by uttamcadambi07** [reply]
Hi please mark as resolved.

We had to source a path which was missing. 

We managed to do it and it compiled fine.
